### PR TITLE
Added ChatroomServer class

### LINKED ISSUE #5: Creating ChatroomServer class

### PR DESCRIPTION
**Pull request by lorenzobucci**
Closes #5

### CODE REVIEW

File: java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        clientSocket.emit("join");
+        clientSocket.emit("msg", originalMessage1.toJSON());
+        clientSocket.emit("msg", originalMessage2.toJSON());
+
+        ArgumentCaptor<Message> argumentCaptor = ArgumentCaptor.forClass(Message.class);
+        await().atMost(2, SECONDS).until(() -> !retrievedMessages.isEmpty());
+        verify(serverRepository, times(2)).save(argumentCaptor.capture());
+        List<Message> capturedArgument = argumentCaptor.getAllValues();
+
+        assertThat(capturedArgument.contains(originalMessage1) && capturedArgument.contains(originalMessage2)).isTrue();
```
Comment (marcodiri) [inline java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java]:
```suggestion
        assertThat(capturedArgument).contains(originalMessage1, originalMessage2);
```

File: java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        clientSocket.on("msg", arg -> {
+            JSONObject jsonMsg = (JSONObject) arg[0];
+            Message incomingMessage = new Message(new Timestamp(jsonMsg.getLong("timestamp")), jsonMsg.getString("user"), jsonMsg.getString("message"));
+            retrievedMessages.add(incomingMessage);
+        });
+        clientSocket.connect();
+        clientSocket.emit("join");
+
+
+        await().atMost(2, SECONDS).until(() -> retrievedMessages.contains(msg1) && retrievedMessages.contains(msg2));
```
Comment (marcodiri) [inline java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java]: Make the reason why the test fails clearer. The same goes for similar `await()` tests.
For the suggestion to work `import static java.util.Arrays.asList;` is needed.
```suggestion
        try {
        	await().atMost(2, SECONDS).until(() -> retrievedMessages.containsAll(asList(msg1, msg2)));
		} catch (org.awaitility.core.ConditionTimeoutException e) {
			fail("Expected " + asList(msg1, msg2) + " but got " + retrievedMessages);
		}
```

**Review by marcodiri — CHANGES_REQUESTED** [review]

File: java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java (hunk 1 above)
Comment (lorenzobucci) [reply java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java]: Solved

File: java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java (hunk 2 above)
Comment (lorenzobucci) [reply java_socketio_chatroom_server/src/it/java/io/github/marcodiri/java_socketio_chatroom_server/ChatroomServerIT.java]: Solved in all tests

**Review by marcodiri — APPROVED** [review]

**Pull request merged**